src/test/java/one/util/streamex/MoreCollectorsTest.java

```diff
@@ -591,6 +594,243 @@ public void testToEnumSet() {
         checkCollectorEmpty("Empty", EnumSet.noneOf(TimeUnit.class), MoreCollectors.toEnumSet(TimeUnit.class));
     }
 
+    /**
+     * See {@link MoreCollectors#entriesToMap()}.
+     */
+    @Test
+    public void testEntriesToMap() {
+        assertThrows(IllegalStateException.class, () ->
+                EntryStream.generate(() -> "a", () -> 1).limit(10).collect(MoreCollectors.entriesToMap()));
+
+        checkCollectorEmpty("entriesToMap", Collections.emptyMap(), MoreCollectors.entriesToMap());
+
+        {
+            Map<Integer, String> expected = EntryStream.of(1, "*", 2, "**", 3, "***", 4, "****", 5, "*****").toMap();
+            Supplier<Stream<Entry<Integer, String>>> stream = expected.entrySet()::stream;
+            checkCollector("entriesToMap", expected, stream, MoreCollectors.entriesToMap());
+
+            streamEx(stream, supplier -> {
+                Map<Integer, String> result = supplier.get().collect(MoreCollectors.entriesToMap());
+                assertEquals("*", result.get(1));
+                assertEquals("**", result.get(2));
+                assertEquals("***", result.get(3));
+                assertEquals("****", result.get(4));
+                assertEquals("*****", result.get(5));
+            });
+        }
+    }
+
+    /**
+     * See {@link MoreCollectors#entriesToMap(BinaryOperator)}.
+     */
+    @Test
+    public void testEntriesToMapWithCombiner() {
+        streamEx(() -> Stream.of(
+                EntryStream.of(1, "*").toMap(),
+                EntryStream.of(1, "*", 2, "*").toMap(),
+                EntryStream.of(1, "*", 2, "*", 3, "*").toMap(),
+                EntryStream.of(1, "*", 2, "*", 3, "*", 4, "*").toMap(),
+                EntryStream.of(1, "*", 2, "*", 3, "*", 4, "*", 5, "*").toMap())
+                .flatMap(map -> map.entrySet().stream()), supplier -> {
+            Map<Integer, String> result = supplier.get().collect(MoreCollectors.entriesToMap(String::concat));
+            assertEquals("*****", result.get(1));
+            assertEquals("****", result.get(2));
+            assertEquals("***", result.get(3));
+            assertEquals("**", result.get(4));
+            assertEquals("*", result.get(5));
+        });
+
+        streamEx(() -> Stream.of(
+                EntryStream.of(1, "one", 2, "two", 3, "three", 4, "four").toMap(),
+                EntryStream.of(1, "ein", 2, "zwei", 3, "drei").toMap(),
+                EntryStream.of(1, "une", 2, "deux").toMap())
+                .flatMap(map -> map.entrySet().stream()), supplier -> {
```

Comment:
Looks like `toMap()` is redundant here. Just create a `Stream<EntryStream>`, then `flatMap(Function.identity())`.